src/components/Fields/TextField.js

```diff
@@ -260,11 +260,12 @@ class TextField extends Component {
     }
 
     render() {
+        const hasValue = this.state.value === 0 ? true : Boolean(this.state.value);
```

Comment:
Approved.

Although, now I'm thinking it would've been more elegant to just write this instead:
`const hasValue = this.state.value === 0 || Boolean(this.state.value);`